### PR TITLE
REL-897911 Added job based workflow samples

### PR DESCRIPTION
**Pull request by KlaudiuszBryjaRelativity**

### CODE REVIEW

File: Source/Relativity.Transfer.SDK.Samples.Core/appsettings.json [hunk 1]
```diff
@@ -58,5 +58,21 @@
...
     // A valid workspace ID (-1 is a default workspace ID).
     "WorkspaceId": -1
+  },
+  "UploadDirectoryBasedOnExistingJob": {
+    // Should be a local directory path.
+    "FirstSource": "",
+    // Should be a local directory path and it should be different than FirstSource.
+    "SecondSource": "",
+    // Should be an UNC format path to the directory on RelativityOne fileshare.
+    "Destination": ""
```
Comment (KamilKalitaRel) [inline Source/Relativity.Transfer.SDK.Samples.Core/appsettings.json:68]: Can explain those params? I thought this kind of transfer needs a source (one) and the ID of a download job to get its source as the destination.

Comment (KlaudiuszBryjaRelativity) [reply Source/Relativity.Transfer.SDK.Samples.Core/appsettings.json:68]: You are right but @mmkowal-rel do it like this:

- Upload:
    - Upload some data to a destination path using job based workflow (so he need to register this transfer using source and destination) - maybe full path workflow should be used here
    - The second step, which is the main part of the sample, is the transfer of some data set to the same destination path. In that case we use job based workflow.
- Download:
    - Upload some data to a destination path using this time full path workflow
    - The second step, which is the main part of the sample, is download that data set to the local destination path. In that case we use job based workflow

File: Source/Relativity.Transfer.SDK.Samples.Core/Configuration/Configuration.cs [hunk 2]
```diff
@@ -40,7 +48,9 @@ internal record Configuration(
 			new SourceAndDestinationConfiguration(string.Empty, string.Empty),
 			new SourceAndDestinationConfiguration(string.Empty, string.Empty),
 			new SourceAndDestinationConfiguration(string.Empty, string.Empty),
-			new SourceAndWorkspaceIdConfiguration(string.Empty, -1));
+			new SourceAndWorkspaceIdConfiguration(string.Empty, -1),
```
Comment (avichay-kardash) [inline Source/Relativity.Transfer.SDK.Samples.Core/Configuration/Configuration.cs:51]: -1 means workspace id?

Comment (KlaudiuszBryjaRelativity) [reply Source/Relativity.Transfer.SDK.Samples.Core/Configuration/Configuration.cs:51]: yes, this is default WorkspaceId (admin)

**Review by KamilKalitaRel — APPROVED** [review]

**Review by avichay-kardash — APPROVED** [review]

**Pull request merged**